packages/@aws-cdk/aws-iotevents/test/detector-model.test.ts

```diff
@@ -0,0 +1,309 @@
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    DetectorModelDefinition: {
+      InitialStateName: 'test-state',
+      States: [{
+        StateName: 'test-state',
+        OnEnter: {
+          Events: [{
+            EventName: 'test-eventName',
+            Condition: 'test-eventCondition',
+          }],
+        },
+      }],
+    },
+    RoleArn: {
+      'Fn::GetAtt': ['MyDetectorModelDetectorModelRoleF2FB4D88', 'Arn'],
+    },
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::IAM::Role', {
+    AssumeRolePolicyDocument: {
+      Statement: [{
+        Action: 'sts:AssumeRole',
+        Effect: 'Allow',
+        Principal: { Service: 'iotevents.amazonaws.com' },
+      }],
+    },
+  });
+});
+
+test('can get detector model name', () => {
+  // GIVEN
+  const detectorModel = new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnter: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // WHEN
+  new cdk.CfnResource(stack, 'Res', {
+    type: 'Test::Resource',
+    properties: {
+      DetectorModelName: detectorModel.detectorModelName,
+    },
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('Test::Resource', {
+    DetectorModelName: { Ref: 'MyDetectorModel559C0B0E' },
+  });
+});
+
+test('can set physical name', () => {
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    detectorModelName: 'test-detector-model',
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnter: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    DetectorModelName: 'test-detector-model',
+  });
+});
+
+test('can set multiple events to State', () => {
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnter: [
+        {
+          eventName: 'test-eventName1',
+          condition: iotevents.Expression.fromString('test-eventCondition'),
+        },
+        {
+          eventName: 'test-eventName2',
+        },
+      ],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    DetectorModelDefinition: {
+      States: [
+        Match.objectLike({
+          OnEnter: {
+            Events: [
+              {
+                EventName: 'test-eventName1',
+                Condition: 'test-eventCondition',
+              },
+              { EventName: 'test-eventName2' },
```

Comment:
I would probably keep the same formatting as on the previous line, otherwise this looks a little jarring:

```suggestion
              {
                EventName: 'test-eventName2',
              },
```